### ISSUE TITLE
About the model performance on NYT-H

### ISSUE THREAD

**Issue by antct** [Spico197/NYT-H#4]
Hi, thanks for your work first.
I have downloaded the NYT-H dataset and try to reproduce the results under Bag2Bag setting.
The data preprocessing process is like that:

For the train.json/dev.json:
a. I transform the format of train.json and dev.json to the format of OpenNRE.

For the test.json:
b. I record the label for each bag. (if the item `bag_label` is yes).
c. I try to recover the label for each instance. (using the label recorded in b), if the `bag_label` of the instance is no)

And I get an AUC metric of 89.3, it seems a little strange. Can your provide more details about it?

**Comment by Spico197** [comment]
- Did you merge `na_train.json` with `train_nonna.json` to generate the final `train.json` file?
- Among all the metrics, which kind of AUC did you use? `auc`, `non_na_auc` or `b_non_na_auc`? (for DSGT scores, `non_na_auc` is used, for MAGT scores, please use `b_non_na_auc` instead)
- Not all bags have `yes` label, do you mean you've shrunk the size of test set?

**Comment by antct** [comment]
Thanks for your reply.

- Yes, I have merged the `na_train.json` and `train_nonna.json` to get the final `train.json`. It has 193,662 lines.
- Just `auc`.
- Yes, I find some bags don't have true labels. For these bag, the recovering process doesn't work and I just skip it.

**Comment by Spico197** [comment]
AUC scores in the paper is the DSGT AUC (`non_na_auc`), if you want to get the MAGT AUC, please use the `b_non_na_auc` scores.

**Comment by antct** [comment]
Sorry, the mertic I use is `non_na_auc`, which excludes the relation `NA` (following the setting in OpenNRE).
I found a similar issue #3 , they also report a high AUC score.
[Wed 09 Dec 2020 15:19:14]-INFO-evaluate.py-347-18063:   non_na_auc = 0.8969445947216447
Maybe the AUC score reported in the paper includes the relation `NA`?

**Comment by Spico197** [comment]
No. The AUC score is not related to  any relation since it's micro-averaged. AUC scores in the paper are `non_na_auc`. It's OK to get higher results than the paper reported, since we have set 5 seeds and take the mean results as the final results. If you wanna reproduce the results, please do not change the test set and check the log below (PCNN+ATT Bag2Bag Seed 422) to follow the hyperparameters.

[pcnn_att_422.log](https://github.com/Spico197/NYT-H/files/5871451/pcnn_att_422.log)

**Comment by Spico197** [comment]
I've just re-run the PCNN+ATT (Bag2Bag) trial, and here is the log. Compared to the experiments results before, it is successfully reproduced.

[pcnn_att_422-reproduce.log](https://github.com/Spico197/NYT-H/files/5871745/pcnn_att_422-reproduce.log)

**Comment by antct** [comment]
Thanks for your reply. I will check it carefully!
